### ISSUE TITLE
cabal configure doesn't work

### ISSUE THREAD

**Issue by ghost** [bangalore-haskell-user-group/functorrent#29]
On running functorrent - cabal in sandbox mode,

```
cabal sandbox init
cabal configure 
```

I got this error,

``` haskell
Resolving dependencies...
Configuring functorrent-0.1.0.0...
cabal: At least the following dependencies are missing:
tasty -any, tasty-hunit -any
```

After this error, doing cabal install --only-dependencies worked for me. But most people do cabal configure, build, install, why are we taking a different route?

**Comment by jaseemabid** [comment]
@madhavan020985 I'm not familiar with cabal intricacies. Did you miss the wget step? We don't track the `cabal.config` file at the moment and it should be root cause for non reproducible builds. I think this can be solved by checking in the latest lts stackage cabal.config. 

@vu3rdd, comments?

**Comment by vu3rdd** [comment]
@madhavan020985 @jaseemabid 
This sounds like the test dependencies are not installed. I believe if you do `cabal configure --disable-tests` if you don't want to see this error or install the ones it is complaining about.

By a 'different route' you mean, usage of `cabal sandbox`? The problem is because we have at least 3 different ghc versions in use (7.4.x, 7.6.x and 7.8.x) and now the new 7.10.x. Also there are multiple versions of each package and often these versions are incompatible with each other (different APIs..) and packages depend on specific versions of packages. Doing a `cabal install' can bring in a lot of packages into your _global_ package cache and that may not be compatible with your other existing haskell program installations that you installed via cabal-install. So, we tend to sandbox package installations. cabal sandbox appeared in only recent versions of cabal-install. It does not fix all problems. We are using stackage as it is a curated set of packages and versions. It needs ghc 7.8 for it to work. 

Another solution I am experimenting with at the moment is to use the [nix package manager](https://nixos.org/nix/). It looks quite promising. Nix solves  a lot of these problems very nicely. But I understand that it is too much to ask people to move to another package manager. Like Haskell, nix has a different take on how it solves common problems and it needs different ways of thinking. I really really like nix at this point.

So, stackage+cabal sandbox seem to be the best solution we have at the moment.

**Comment by ghost** [comment]
@jaseemabid , i did do wget. but i did, "cabal configure" after wget. it is not there in your prescribed list of steps. i did it, because for all hackage source code, i do cabal configure, cabal build & cabal install. i think, for sandbox installation, the procedure can be different. in btw, is --only-dependencies option meant to tell cabal to put its downloaded binaries inside the sandbox and not in the global folders?

**Comment by vu3rdd** [comment]
@madhavan020985 No, the cabal install --only-dependencies will install the dependencies in your sandbox only if you are invoking it inside a sandbox. If you do 'cabal sandbox delete' and then invoke 'cabal install --only-dependencies', then the dependencies are installed globally.
